Sources/TuistKit/Services/Lint/LintCodeService.swift

```diff
@@ -0,0 +1,135 @@
+    private let manifestLoading: ManifestLoading
+
+    init(rootDirectoryLocator: RootDirectoryLocating = RootDirectoryLocator(),
+         codeLinter: CodeLinting = CodeLinter(),
+         manifestLoading: ManifestLoading = ManifestLoader(),
+         graphLoader: GraphLoading = GraphLoader(modelLoader: GeneratorModelLoader(manifestLoader: ManifestLoader(),
+                                                                                   manifestLinter: AnyManifestLinter())))
+    {
+        self.rootDirectoryLocator = rootDirectoryLocator
+        self.codeLinter = codeLinter
+        self.manifestLoading = manifestLoading
+        self.graphLoader = graphLoader
+    }
+
+    func run(path: String?, targetName: String?) throws {
+        // Determine destination path
+        let path = self.path(path)
+
+        // Load graph
+        let graph = try loadDependencyGraph(at: path)
+
+        // Get sources
+        let sources = try getSources(targetName: targetName, graph: graph)
+
+        // Lint code
+        logger.notice("Running code linting")
+        try codeLinter.lint(sources: sources, path: path)
+    }
+}
+
+// MARK: - Destination path
+
+private extension LintCodeService {
+    func path(_ path: String?) -> AbsolutePath {
+        guard let path = path else { return FileHandler.shared.currentPath }
+
+        return AbsolutePath(path, relativeTo: FileHandler.shared.currentPath)
+    }
+}
+
+// MARK: - Load dependency graph
+
+private extension LintCodeService {
+    func loadDependencyGraph(at path: AbsolutePath) throws -> Graph {
+        let manifests = manifestLoading.manifests(at: path)
+
+        logger.notice("Loading the dependency graph")
+        if manifests.contains(.workspace) {
+            logger.notice("Loading workspace at \(path.pathString)")
+            let (graph, _) = try graphLoader.loadWorkspace(path: path)
+            return graph
+        } else if manifests.contains(.project) {
+            logger.notice("Loading project at \(path.pathString)")
+            let (graph, _) = try graphLoader.loadProject(path: path)
+            return graph
+        } else {
+            throw LintCodeServiceError.manifestNotFound(path)
+        }
+    }
+}
+
+// MARK: - Get sources to lint
+
+private extension LintCodeService {
+    func getSources(targetName: String?, graph: Graph) throws -> AbsolutePath {
+        if let targetName = targetName {
+            return try getTargetSources(targetName: targetName, graph: graph)
+        } else {
+            return graph.entryPath
+        }
+    }
+
+    // TODO: move it to Graph extension (?)
+    func getTargetSources(targetName: String, graph: Graph) throws -> AbsolutePath {
+        guard let target = graph.targets.flatMap({ $0.value }).map(\.target).first(where: { $0.name == targetName }) else {
+            throw LintCodeServiceError.targetNotFound(targetName)
+        }
+
+        // TODO: move it to AbsolutePath+Extras.swift (?)
```

Comment:
@fortmarek do you still think that it should be moved to `AbsolutePath+Extras.swift`

Reply:
I think it would be a nice extension for `Collection`, but it's not necessary - it just *might* be useful in the future